examples/tests/valid/capture_resource_with_no_inflight.w

```diff
@@ -0,0 +1,13 @@
+bring cloud;
+
+// This resource has no inflight methods (no client) but it still needs to be captured
+// so we can access its fields.
+resource A {
+  field: str;
+  init() { this.field = "hey"; }
+}
+
+let a = new A();
+new cloud.Function(inflight () => {
+ assert("hey" == a.field);
+}) as "test";
```

Comment:
Do we have a test where we access a preflight field of a resource that does have inflight methods? (Just to cover that case as well)

Reply:
That's **is** test. This is what this PR fixes!
There are two different way to get this done: what I did in this PR and a more complex way but possibly better that's mentioned in the top comment.
In any case this test is a test of accessing a preflight field of a resource without inflight methods.